CRM/Contact/BAO/Query.php

```diff
@@ -4749,7 +4748,7 @@ public static function getGroupByFromSelectColumns($selectClauses, $groupBy = NU
           $primaryKey = "{$tableName}.id";
           // exclude columns which are already included in groupBy and aggregate functions from select
           // CRM-18439 - Also exclude the columns which are functionally dependent on columns in $groupBy (MySQL 5.7+)
-          if (!in_array($selectColumn, $groupBy) && !in_array($primaryKey, $groupBy) && preg_match($regexToExclude, trim($selectColumn)) !== 1) {
```

Comment:
Even tho we do group by on primary key need to group by on other columns as well for full_group_by to work